contracts/src/Gateway.sol

```diff
@@ -394,6 +416,49 @@ contract Gateway is IGateway, IInitializable {
         emit PricingParametersChanged();
     }
 
+    // @dev Register a new fungible Polkadot token for an agent
+    function _registerForeignToken(bytes32 agentID, address agent, bytes memory params) internal {
```

Comment:
With the new approach, I was thinking that the `AgentExecuteCommand.RegisterToken` command would be replaced by a toplevel `Command.RegisterToken` command that is dispatched in `submitV1`.

Also all these new asset-specific handling functions need to be moved to `Assets.sol` to keep within the contract size limit.

Reply:
https://github.com/Snowfork/snowbridge/pull/1155/commits/fdf15aa92abe3bd9d35ce8d21f2f62e8f29cd4df and https://github.com/Snowfork/polkadot-sdk/pull/128/commits/fec5a63c8d24669928986fc749e2be5e5b1365b6